File: move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int count = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] == 0) {
                count++;
            }
        }

        int count2 = 0;
        for (int i = 0; i < nums.length && count2 < nums.length - count; i++) {
            if (nums[i] != 0) {
                nums[count2] = nums[i];
                count2++;
            }
        }

        for (int i = count2; i < nums.length; i++) {
            nums[i] = 0;
        }
    }
}

File: 283-move-zeroes/move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        slow = 0
        count = 0
        for fast in range(len(nums)):
            if nums[fast] == 0:
                count += 1
            else:
                nums[slow] = nums[fast]
                slow += 1
        while count:
            nums[slow] = 0
            slow += 1
            count -= 1